S2S_added sample after sharing a case/contact does not get shared [2]

<!--
Please read the Contributing guidelines (https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) before submitting an issue. You don't have to remove this comment or any other comment from this issue as they will automatically be hidden.
-->
### Bug Description
<!-- Mandatory -->
Added sample after sharing a case/contact does not get shared.
### Steps to Reproduce
<!-- Optional; please add more steps if necessary -->
1. Create a case/ contact in source system
2. Add a Sample
3. Share case/contact to target system
4. Add another sample in source system
5. Accept case/contact in target system
6. Check if both samples are visible in target system
7. Currently just one sample is visible in target system
8. Check if both samples are visible in source system
9. Currently both samples are visible in source system

-> tested with different combinations of dates -> no difference, both samples are never visible in target system
-> with vaccinations everything is fine and works

### Expected Behavior
<!-- Optional -->
Both samples should also be visible in target system
### Screenshots
<!-- Optional -->
source system:
![image](https://user-images.githubusercontent.com/100844689/177795237-fd592621-a1e9-4b03-9484-34b80ad83674.png)

target system:
![image](https://user-images.githubusercontent.com/100844689/177795378-1e89f585-23e1-4392-bb22-0b6bc2ee9afb.png)
### System Details
<!-- Mandatory; you only have to specify the Server URL if the error appeared on a publicly available test server -->
* Device: Windows 11
* SORMAS version: 1.73.1
* Android version/Browser: Edge
* Server URL: release-sormas-x / release-de
* User Role: admin

### Additional Information
<!-- Optional -->

Reported by @fhauptmann: Deployment on test-de1 not working:

```sql

Updating database to version 477... (de.symeda.sormas.backend.common.StartupShutdownService.java:738)
ERROR: column "datatype" contains null values

  A system exception occurred during an invocation on EJB UpdateQueryTransactionWrapper, method: public int de.symeda.sormas.backend.common.StartupShutdownService$UpdateQueryTransactionWrapper.executeUpdate(de.symeda.sormas.backend.common.StartupShutdownService$UpdateQueryTransactionWrapper$TargetDb,java.lang.String)]]
sormas_1             |
[2022-08-01T10:32:12.203+0200] [Payara 5.2021.10] [WARNUNG] [] [javax.enterprise.ejb.container] [tid: _ThreadID=114 _ThreadName=payara-executor-service-scheduled-task] [timeMillis: 1659342732203] [levelValue: 900] [[
sormas_1             |
javax.ejb.EJBException: org.hibernate.exception.ConstraintViolationException: could not execute statement
...
 at com.sun.proxy.$Proxy1141.executeUpdate(Unknown Source)
 at de.symeda.sormas.backend.common.__EJB31_Generated__StartupShutdownService$UpdateQueryTransactionWrapper__Intf____Bean__.executeUpdate(Unknown Source)
 at de.symeda.sormas.backend.common.StartupShutdownService.updateDatabase(StartupShutdownService.java:739)
 at de.symeda.sormas.backend.common.StartupShutdownService.startup(StartupShutdownService.java:207)
 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
...
 at org.hibernate.query.internal.AbstractProducedQuery.executeUpdate(AbstractProducedQuery.java:1696)
 ... 133 more
Caused by: org.postgresql.util.PSQLException: ERROR: column "datatype" contains null values
 at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2675)
...
 at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
 ... 137 more
]]
```

--- a/sormas-backend/src/main/resources/sql/sormas_schema.sql
+++ b/sormas-backend/src/main/resources/sql/sormas_schema.sql
@@ -11715,6 +11715,7 @@ UPDATE sharerequestinfo sr SET datatype = (
        WHEN (EXISTS(SELECT caze_id FROM sormastosormasshareinfo s JOIN sharerequestinfo_shareinfo ss ON ss.sharerequestinfo_id = r.id WHERE s.id = ss.shareinfo_id AND caze_id IS NOT NULL)) THEN 'CASE'
        WHEN (EXISTS(SELECT contact_id FROM sormastosormasshareinfo s JOIN sharerequestinfo_shareinfo ss ON ss.sharerequestinfo_id = r.id WHERE s.id = ss.shareinfo_id  AND contact_id IS NOT NULL)) THEN 'CONTACT'
        WHEN (EXISTS(SELECT event_id FROM sormastosormasshareinfo s JOIN sharerequestinfo_shareinfo ss ON ss.sharerequestinfo_id = r.id WHERE s.id = ss.shareinfo_id  AND event_id IS NOT NULL)) THEN 'EVENT'
+       ELSE 'CASE' -- hardcode CASE for share request with no shared object due to permanent deletions
     END
     FROM sharerequestinfo r where r.id = sr.id
 );

